ramp4-pcar4/ramp4-pcar4
# Add ability to register and update panels using plain html

### Related Item(s)
#2242

### Changes
- Added api method for registering a panel using HTML content
- Added api method for updating a panel's content using HTML content
- Modified the type definition of `PanelConfigScreens` to include objects of type `HTMLElement` as values
- In `registerScreens()`, handled the case where the screen is of type `HTMLElement`

### Notes
- When calling `register()` within `registerHTML()`, I have set `alertName` to either be user defined or a default of `panels.html`, which is a key in src/lang/lang.csv. If the user-defined value for `alertName` isn't a key in lang.csv, it will display as plain text in the panel. Should we allow this, or have `alertName` default to `panels.html` if it doesn't exist in lang.csv?
- Panels created in src/fixtures/legend/index.ts are for testing, and will be removed before merging
- Image added in demos is for testing, and will be removed before merging

### Testing
Steps:
1. Open sample 1
2. Observe that, along with the legend panel, 3 additional panels are visible 
3. Observe the first panel has a height and width that is smaller than the others, is colored orange, has a title of 'HTML Element' (which corresponds to the key `panels.html` in src/lang/lang.csv), and has content text of 'creating a panel with styling'
4. Observe the second panel has title 'panel '#'2' (which doesn't correspond to a key and thus displays as plain text) and content text of 'updating panel with an id'
5. Observe the third panel has title 'English' (which corresponds to the key `lang-en` ) and a picture of the earth as content
6. Observe that you can pin/unpin, move left/right, and close these new panels 
7. Steps 1-6 should work for (most) other samples as well
